### ISSUE TITLE
IMAP / SMTP Setup not working

### ISSUE THREAD

**Issue by brunogaspar** [nylas/N1#47]
When i try to add an email account it always says that i need to check the credentials.

I'm pretty sure that the credentials that i'm inputting are 100% correct, since i use them on Postbox (which auto detects these) and even on Airmail, and they work perfectly fine, without any hiccups.

Am i missing something?

Regards!

> Sorry if this was issued before, but i couldn't find any issue by searching.

**Comment by colby-swandale** [comment]
Your going to need to provide a LOT more information than just saying its not working.

**Comment by brunogaspar** [comment]
What kind of information do you require?

As i stated on the OP

>  it always says that i need to check the credentials.

**Comment by colby-swandale** [comment]
What provider/imap server are you using? Are you using SSL? Are you using plaintext passwords or encrypted of some sort? just to name a few.

**Comment by lutzee** [comment]
Hi, I have also tried to setup imap/smtp on N1 till it decided not to try and connect to my local sync-engine instance and I gave up. I can give a little light though. 

I run my own mail server, uses postfix + courier imap. The imap credentials passed fine but when it got to SMTP it wouldn't authenticate. I have my SMTP server set up to use STARTTLS and at no point did the sync-engine even attempt to contact my mail server. 

p.s. At the end of the day the client looks pretty but having a MITM between my mail client and my mail server just isn't worth it. Especially if I'd be forced to bounce my email through a third party server. Better look next time.

**Comment by bengotow** [comment]
Hey @brunogaspar — we've tested the IMAP support with a range of different settings (under the hood, we use our IMAP support for Yahoo / AOL / iCloud, etc.) Could you give me your connection settings sans any sensitive info (username/password)? I can create an issue for our backend team to investigate and we'll figure out what the issue is.

**Comment by brunogaspar** [comment]
@bengotow Sorry for the late reply.

I've tried with another email account, and it's working. But not entirely sure why it's not working for the other email accounts since they work perfectly fine on other email clients.

I can create a temporary email account and give you the credentials so you can help us debug what might be wrong.

Both servers have cPanel.

Let me know how you want to proceed.

**Comment by gardner** [comment]
@brunogaspar can you reproduce this with a temporary email account?

**Comment by wesdegroot** [comment]
Same problem here.
 `{"message": "Please check that your SMTP settings are correct.", "type": "auth_error", "error": "further_action_required"}`

the SMTP server working on Apple mail.

Some details (except password ofc.)
`curl -X POST -d '{"settings":{"imap_port":993,"imap_ssl_enabled":false,"smtp_port":587,"smtp_ssl_enabled":false,"imap_host":"wdgwv.nl","imap_username":"wes@wdgwv.nl","imap_password":"MYPASSWORD","smtp_host":"wdgwv.nl","smtp_username":"wes@wdgwv.nl","smtp_password":"MYPASSWORD"},"name":"Wesley de Groot","email":"wes@wdgwv.nl","provider":"imap"}' "https://:@api.nylas.com/auth?client_id=MYCLIENTID"`

What the weird part about it is, is that in the GUI the smtp port is set to 25. (like normal SMTP's do)
If i'm running from the command line itself i got the same message.

Even if i manually set the smtp to 25 (in command line test)

So, i can't see anything, unless the Api send us more information.

**Comment by grinich** [comment]
Folks, I think this is related to our backend system that uses imapclient. We should now be able to support STARTLS though, right @mjs ?

**Comment by peperami97** [comment]
I have version 0.3.19 and this is still not working this morning.

**Comment by vkorecky** [comment]
Hi,
I have a similar problem.
SMTP: without SSL, port 25, annonymous authentication
Error message: Please provide your password and username

**Comment by geroyche** [comment]
not working for me either, for a custom imap setup with ssl and old certificates.
"Please check that your SMTP settings are correct."

works in Thunderbird, but only after i accept weak dh via config tweak, and works in gmail android client when i say "accept all certificates".
maybe there's an "issue" there with nylas being "too strict"?

**Comment by shinigammi** [comment]
dont work for me the SMTP, the same problem, the IMAP mail is in godaddy, this data maybe help

**Comment by abildma** [comment]
same problem here. 

It works in Thunderbird after accepting weak certificates

**Comment by thaapaniemi** [comment]
Same problem here with self-hosted smtp server (citadel) and self-signed certficates. from log (mail.err):
> Oct 28 23:25:20 Server citserver[14625]: SSL_accept failed: retval=0, errval=1, err=error:00000001:lib(0):func(0):reason(1)
> Oct 28 23:25:21 Server citserver[14625]: SMTP: client disconnected: ending session.

**Comment by semsphy** [comment]
+1

**Comment by thaapaniemi** [comment]
Update: I got N1 IMAP/SMTP working by using signed SSL certificates so problem was (at least for me) self-signed cert.

**Comment by xeaon** [comment]
there should be an option to use self-signed certificates imho

**Comment by visualcookie** [comment]
:+1: Would be great to see, where the configuration was wrong instead of just showing the information which isn't helping the user. I double checked my configuration with Apple Mail and it all was correct, but still get the same error which is not helping me much.

**Comment by sdrap** [comment]
Apparently this is the same for me. SMTP with signed certificate is ok, but SMTP with self signed ones no. Using msmtp with mutt works with smtp 587 "auth login" as option.

**Comment by jukben** [comment]
+1 
Same problem here. I exactly rewrite configuration from Geary (where everything works fine) and I got error "Please check that your SMTP settings are correct."

**Comment by roobre** [comment]
I think I'm having the same issue with TLS and self-signed certs.

In case it heps, my postfix says:

````
Nov 04 03:48:53 Tenshi postfix/smtpd[57136]: connect from ec2-52-32-74-203.us-west-2.compute.amazonaws.com[52.32.74.203]
Nov 04 03:48:54 Tenshi postfix/smtpd[57136]: SSL_accept error from ec2-52-32-74-203.us-west-2.compute.amazonaws.com[52.32.74.203]: 0
Nov 04 03:48:54 Tenshi postfix/smtpd[57136]: warning: TLS library problem: error:14094418:SSL routines:SSL3_READ_BYTES:tlsv1 alert unknown ca:s3_pkt.c:1294:SSL alert number 48:
Nov 04 03:48:54 Tenshi postfix/smtpd[57136]: lost connection after STARTTLS from ec2-52-32-74-203.us-west-2.compute.amazonaws.com[52.32.74.203]
Nov 04 03:48:54 Tenshi postfix/smtpd[57136]: disconnect from ec2-52-32-74-203.us-west-2.compute.amazonaws.com[52.32.74.203]
````

**Comment by seitzbg** [comment]
Still doesn't work here either.  Thunderbird works fine.

**Comment by CapitalH** [comment]
I have not yet had a chance to make N1 connect to my SMTP account. I also get the "Please check that your SMTP settings are correct." error all the time. This is the only drawback for me to fully move all my email work to N1.

I hope this gets fixed soon.

**Comment by golimpio** [comment]
Similar issue. I've tested several email clients (Thunderbird, Geary Mail, Apple Mail, Mutt) and N1 is the only one that's not working so far.

Following my settings from Geary Mail:

*IMAP Settings:*
Server: `mail.company.com.au`
Port: `993`
User: `first last@company.ho`
Password: `???`
Encryption: `SSL/TLS`

*SMTP Settings:*
Server: `mail.company.com.au`
Port: `587`
User: `first last`
Password: `???`
Encryption: `STARTTLS`

Probably it's not important, but my username is my _first_ and _last name_, including the space between them.

**Comment by rista404** [comment]
Hi, similar problem. Maybe my data can help :)

```
Secure SSL/TLS Settings

Username:	something@domain.com
Password:	pass
Incoming Server:	server4.something.com
IMAP Port: 993
POP3 Port: 995
Outgoing Server:	server4. something.com
SMTP Port: 465
```

Authentication is required for IMAP, POP3, and SMTP.

Cheers!

**Comment by seitzbg** [comment]
Is there going to be a fix for this so the product is actually usable ? :)

**Comment by dasDaniel** [comment]
same issue, server using TLS on port 587 for SMTP

**Comment by lmartins** [comment]
Was hoping this last update would solve this, unfortunately that isn't the case. It's pretty early days for N1, so Im still quite confident in it's future, but please fix this so we can help testing it out. Thanks.

**Comment by cozo** [comment]
Hello,

I don t really understand your strategy with code for trying an "open source" software, i think is for making the "buzz". But i DL N1 yesterday and i have the same problèm it just don't work with smtp with starttls (my own server).
And without repo how can i manage update ? 
Shall i request a new code ....
just put a repo and let everybody try, n1 seems to be really nice.

**Comment by kevin39** [comment]
Same here... Own server (Axigen Server).

**Comment by madfriend** [comment]
SMTP still not working (internal work server, so can't really provide more info)

**Comment by visualcookie** [comment]
Started it in Developer-Mode now. Got the following error:
```
[SECURITY] node-uuid: crypto not usable, falling back to insecure Math.random()
```

Checked my login credentials and they're correct. Can we have a fix here anytime soon? I really want to give N1 a try, but with the current state it's not possible.

**Comment by nicolasrenon** [comment]
Hello.
I wasn't able to add my IMAP account as well, after setting up the SMTP info the submit button was loading without any result. But after restarting the app I had the surprise to see that my account was finally set up!
Don't know how to provide data to help.
My IMAP account is a Fastmail email address.

**Comment by EthanBlackburn** [comment]
We've added support for self-signed SSL certificates in https://github.com/nylas/sync-engine/commit/c5981d5a3084a04e97a0d73e66a64ab3ad63460d

**Comment by lmartins** [comment]
Im assuming this is not in build 0.3.25 because using that version I still get the same error.

**Comment by CapitalH** [comment]
Me2, I am still getting the same error as before after updating N1.
If there are some steps we should follow to fix it, please share with us.

**Comment by ashnur** [comment]
I am hugely excited about N1, but I am unable to even add the account because it doesn't even have the most basic auth features other clients(thunderbird, sylpheed, gyazmail) have. 

I am not sure why would anyone build a software which fails for anyone whose email servers aren't set up in the one and only way the software's authors think is "enough"?

I need unencrypted, TLS, STARTTLS, SSL all versions... I mean sure, there is some which is seriously leaky and unsecure, but that doesn't mean it shouldn't be supported at all. Even my browser let me access websites which have self-signed certificates..

**Comment by visualcookie** [comment]
> I am not sure why would anyone build a software which fails for anyone whose email servers aren't set up in the one and only way the software's authors think is "enough"?

@ashnur Because it's beta.

I'm still not able to add my companies email account though. We use an unsigned certificate, hope that helps you with fixing.

**Comment by TUBii** [comment]
Same here. Just tried to connect with the same settings as my Thunderbird uses, and got SMTP error. Thunderbird was configured automatically by domain provider and these settings are the sam what was provided in mail configuration page in domain hosting.

**Comment by polarity** [comment]
me too. console.log:

    [SECURITY] node-uuid: crypto not usable, falling back to insecure Math.random()

my smtp needs:

    Relaying via SMTP AUTH 587 (SMTP Submit) with STARTTLS (RFC 4409);

**Comment by ashnur** [comment]
@visualcookie fair enough, however it should be then made clear to the early adopters what features are and aren't implemented, so we do not waste time trying out something which lacks the minimum features we need to even install it.

**Comment by pepa65** [comment]
Tried 13 different smtp accounts, it's not working, I give up.

**Comment by hughwilliams94** [comment]
Although the recent update has resolved one of my IMAP accounts, I still consistently receive request time out errors when adding an IMAP account from my university, it uses an internal server and plain text pass and ssl for both IMAP and SMTP. What more details are required to help me troubleshoot?

**Comment by xeaon** [comment]
All IMAP Servers you want to connect to must have valid SSL certificates. Check the sync engine source code.

**Comment by hughwilliams94** [comment]
I thought that self-signed certs were now accepted?

**Comment by visualcookie** [comment]
@xeaon So N1 won't be usable for people who haven't signed their certificates? This sucks.

**Comment by xeaon** [comment]
@visualcookie i've testet N1+Sync Engine couple weeks ago. I don't follow the commits anymore, so i don't know if anything has changed.

I would never connect to my mail servers through a third party and running the sync engine to be able to use N1 as 'standonle' mail client is a bit too much. :)

**Comment by YarmoM** [comment]
Same here, university email address gets refused for "Invalid credentials". Trust me, they aren't. How is this possible? Come on you guys, N1 deserves a breakthrough, but this ain't the way you're gonna get it...

**Comment by andratwiro** [comment]
+1, I can configure my SMTP account without SSL, but event then it doesn't allow sending email.

**Comment by xclabit** [comment]
+1, my university email account is also getting "Invalid Credentials" using fairly standard IMAP/SMTP settings. It requires password authentication (yes I know, insecure) but there's no option for it. The same credentials work on clients (Mail.app, iOS mail) with this explicit option. Funny enough, the Windows 10 Mail app has a similar problem.

**Comment by afturner** [comment]
+1 a HUGE issue. Three email accounts under Webfaction using the SSL information.. it works everywhere else, but here it gives me an "Invalid Credentials" warning. 

This must be fixed.

**Comment by bengotow** [comment]
Hey folks—please check out the limitations of our IMAP/SMTP support here: https://nylas.zendesk.com/hc/en-us/articles/216118977-What-are-the-limitations-of-IMAP-SMTP-support-

There are a couple things we don't plan to support, including IMAP/SMTP servers on private networks and servers that send passwords in plaintext. The "username must be your email address" restriction is one we'll probably fix in the near future.

**Comment by afturner** [comment]
@bengotow so any thoughts on why the webfaction smtp doesn't work? It meets all the requirements.

**Comment by willemarcel** [comment]
@afturner I couldn't authenticate in the webfaction smtp too

**Comment by kavmi** [comment]
I have same problem. I'ts working with Outlook acc but fail with  SMTP AOL "Error conecting to the server" and I'm sure I'ts all right.

**Comment by Matsumot0** [comment]
Same problem here via imap servers of my provider @free.fr
:-1:

**Comment by softcoder** [comment]
Same here, pop and smtp via powweb.com says it cannot connect to server, yet works fine in thunderbird.

This host uses connection security: STARTTLS on port 110 for pop.powweb.com
and smtp.powweb.com port 587

Full details here: http://www.powweb.com/knowledgebase/beta/article.bml?ArticleID=180

**Comment by lemmon** [comment]
I used to have the same problem. Then I tried to disable country restriction settings at my mail/smtp provider and Nylas added my account without any issue. Apparently it calls some service that checks SMTP settings outside my region.

**Comment by tigelbri** [comment]
+1 for smtp settings are incorrect - on host freenet (mx.freenet.de)

**Comment by tallcoder** [comment]
+1 

I am having the same problem. My email account used to be with Google, but has since moved to a normal IMAP provider which uses STARTTLS. I even tried deleting my account and creating a new one with no luck. Tried SSL and without and confirmed that all settings are correct.

**Comment by golimpio** [comment]
I don't think they will ever fix this. I was never able to use it because of this specific issue.

**Comment by AndrueCope** [comment]
Yah, not working for me with my personal server either. K9 is working (but the Android 6.0.1 breaks notifications). Google mail won't even let me get past trying to setup IMAP. Just bleats that the username or password is incorrect.

**Comment by nativeimaging** [comment]
I was getting duplicates of all my emails, not sure the cause, but after trying to re-establish my hosted email IMAP I can't event connect back again. Is this issue fixed? Or is it determined as user error?

**Comment by chakrihacker** [comment]
Same here

**Comment by HermannHH** [comment]
Same issue. Had to switch to another client because of this.

**Comment by Teisi** [comment]
+1

**Comment by edgareler** [comment]
+1

Same here: `Incorrect SMTP username or password`. Same settings are working on other clients.

**Comment by kaidiwang** [comment]
+1

**Comment by john-landgrave** [comment]
+1

This app had been working fine for weeks and this just started happening with my connected Google Accounts. I tried hitting the "Re-Authenticate with Google" button in preferences under Accounts, and doing that combined with a full restart seemed to fix it for a little while. I can't help but notice that this has, for me, coincided with the launch of Google's new Sign-In page; any possible connection there?

**Comment by shikkaba** [comment]
+1 Having this problem with Outlook.com
The SMTP username and password are correct.

> Incorrect SMTP username or password.
> Some providers require an app password. Learn more.

**Comment by GeneralCan** [comment]
+1 to this

**Comment by bardogpierre** [comment]
+1

**Comment by VinnyFonseca** [comment]
+1 for outlook, yahoo, and godaddy's workspace

**Comment by karelit0** [comment]
+1 is not working on hotmail account neither on private smtp account.

**Comment by bardogpierre** [comment]
Left and went to Polymail. Needed to connect two email's with GoDaddy's workspace. The UX isn't as clean, as Nylas, but 90% of the functionality is there + support for far more IMAP/SMTP providers.

**Comment by CarnageEmm** [comment]
I am having problems to connect too,  my server are in the domain of xolo.conabio.gob.mx

**Comment by bengotow** [comment]
Hey folks—if you're still having trouble getting your email account to work with Nylas Mail, you should try the new [Mailspring](https://getmailspring.com/) client. Nylas sunset Nylas Mail back in August and I've been working on Mailspring since then. Instead of Javascript-based email sync, it uses Mailcore2 (the same IMAP and SMTP library used by Airmail and Sparrow), and I've found that the switch fixed a lot of connection issues with different mail providers. Hope that helps!

**Comment by vnpnlz** [comment]
@bengotow thanks for the [Mailspring](https://getmailspring.com/) and it works fine with the IMAP and SMTP plus the messages rule is also working good.

**Comment by MarkMacheta** [comment]
+1
Still having this issue when moving out of EMClient. Had to install and use Thunderbird.

**Comment by nfntp** [comment]
Hi @MarkMacheta, I struggled a bit first before landing on the right settings. If you want to share more about your current settings, I'd be happy to help troubleshoot. Mailspring (formerly Nylas) is one of the better mail clients out there.
